SensorApplication.java

```diff
@@ -0,0 +1,161 @@
+        radiationC.weightx = .5;
+        radiationC.anchor = GridBagConstraints.LINE_START;
+        radiationBar.setPreferredSize(new Dimension(warningBarWidth[1], 100));
+        radiationPnl.add(radiationBar, radiationC);
+        radiationPnl.add(new JLabel(output[1]), radiationC);
+        radiationBar.setVisible(true);
+
+        // Set up temperature sensor section
+        JPanel temperaturePnl = new JPanel(new GridBagLayout());
+        GridBagConstraints temperatureC = new GridBagConstraints();
+        temperaturePnl.setBorder(new TitledBorder("Temperature"));
+        add(temperaturePnl);
+
+        // Manage temperature warning bar -- pressure formatting comments apply
+        JPanel temperatureBar = new JPanel();
+        temperatureBar.setBackground(reportColor[2]);
+        temperatureC.weightx = .5;
+        temperatureC.anchor = GridBagConstraints.LINE_START;
+        temperatureBar.setPreferredSize(new Dimension(warningBarWidth[2], 100));
+        temperaturePnl.add(temperatureBar, temperatureC);
+        temperaturePnl.add(new JLabel(output[2]), temperatureC);
+        temperatureBar.setVisible(true);
+
+        // Load GUI
+        setPreferredSize(new Dimension(WINDOW_SIZE, WINDOW_SIZE));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        setResizable(false); // Window cannot be resize - important for warning bar proportions
+        pack();
+    }
+
+    public static void main(String[] args) {
+        SensorApplication app = new SensorApplication();
+
+    }
+
+    // Based on status of sensor report, returns the corresponding color
+    public Color intepretReport(String report){
+
+        switch(report){
+            case "OK":
+                return Color.GREEN;
+
+            case "CRITICAL":
+                return Color.YELLOW;
+
+            case "DANGER":
+                return Color.RED;
+
+        }
+        throw new InvalidParameterException();
+    }
+
+    // Based on the current value of the sensor, and the minimum value for a "DANGER" status, determines the proportion of the width of the GUI window for the warning window to span
+    // The minimum designation for danger is chosen for the "100%" mark on the bar to allow for a reasonable proportion up to that point, and prevent any overflow for enormous numbers
+    public int interpretValue(double value, double max){
+
+        // If the value is above the minimum DANGER classification, have the warning bar span the entire designated space
+        if (value >= max){
+            return MAX_BAR_WIDTH;
+
+        // If the value is zero, have the warning bar span a small amount to remain visible
+        }  else if (value == 0) {
+            return 2;
+
+        // The warning bar will span a proportion of the current sensor value to the minimum danger classification
+        } else {
+            return (int) (value / max * MAX_BAR_WIDTH);
+        }
+    }
+}
```

Comment:
I like how modular and clean this code is! It also runs very well. No complaints. :)